pipeline/pipeline.go

```diff
@@ -0,0 +1,55 @@
+package pipeline
```

Comment:
Before I review, could we have a package comment to describe what this package is doing/ provides? Not really sure I understand its use case tbh. It might help with the review :smile:

Reply:
Added some docs but still is hard to explain 😆